GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -44,13 +44,39 @@ public UpgradeVerb()
             HelpText = "Pass in this flag to actually install the newest release")]
         public bool Confirmed { get; set; }
 
+        [Option(
+            "dry-run",
+            Default = false,
+            Required = false,
+            HelpText = "Display progress and errors, but don't install GVFS")]
+        public bool DryRun { get; set; }
+
+        [Option(
+            "no-verify",
+            Default = false,
+            Required = false,
+            HelpText = "Don't verify authenticode signature of installers")]
+        public bool NoVerify { get; set; }
+
         protected override string VerbName
         {
             get { return UpgradeVerbName; }
         }
 
         public override void Execute()
         {
+            if (this.DryRun)
+            {
+                if (this.Confirmed)
+                {
+                    this.ReportErrorAndExit(this.tracer, ReturnCode.GenericError, "--dry-run and --confirm arguments are not compatible.");
+                }
+                else
+                {
+                    this.Confirmed = true;

```

Comment:
I'm confused why we are setting this to `true`.  Will you explain this please?

Reply:
+1 -- let's figure out a cleaner way to do this.

Reply:
agree. have tried to simplify please take a look.